taskcluster/client.py

```diff
@@ -2,231 +2,24 @@
-        return data
-
-    def buildUrl(self, methodName, *args, **kwargs):
-        entry = None
-        for x in self._api['entries']:
-            if x['name'] == methodName:
-                entry = x
-        if not entry:
-            raise exceptions.TaskclusterFailure(
-                'Requested method "%s" not found in API Reference' % methodName)
-        apiArgs = self._processArgs(entry, *args, **kwargs)
-        route = self._subArgsInRoute(entry, apiArgs)
-        return self.options['baseUrl'] + '/' + route
+_defaultConfig = copy.deepcopy(config)
 
-    def buildSignedUrl(self, methodName, *args, **kwargs):
-        """ Build a signed URL.  This URL contains the credentials needed to access
-        a resource."""
 
-        if 'expiration' in kwargs:
-            expiration = kwargs['expiration']
-            del kwargs['expiration']
-        else:
-            expiration = self.options['signedUrlExpiration']
-
-        expiration = int(time.time() + expiration)  # Mainly so that we throw if it's not a number
-
-        requestUrl = self.buildUrl(methodName, *args, **kwargs)
-
-        if not self._hasCredentials():
-            raise exceptions.TaskclusterAuthFailure('Invalid Hawk Credentials')
-
-        clientId = utils.toStr(self.options['credentials']['clientId'])
-        accessToken = utils.toStr(self.options['credentials']['accessToken'])
-
-        def genBewit():
-            # We need to fix the output of get_bewit.  It returns a url-safe base64
-            # encoded string, which contains a list of tokens separated by '\'.
-            # The first one is the clientId, the second is an int, the third is
-            # url-safe base64 encoded MAC, the fourth is the ext param.
-            # The problem is that the nested url-safe base64 encoded MAC must be
-            # base64 (i.e. not url safe) or server-side will complain.
-
-            # id + '\\' + exp + '\\' + mac + '\\' + options.ext;
-            resource = mohawk.base.Resource(
-                credentials={
-                    'id': clientId,
-                    'key': accessToken,
-                    'algorithm': 'sha256',
-                },
-                method='GET',
-                ext=utils.toStr(self.makeHawkExt()),
-                url=requestUrl,
-                timestamp=expiration,
-                nonce='',
-                # content='',
-                # content_type='',
-            )
-            bewit = mohawk.bewit.get_bewit(resource)
-            return bewit.rstrip('=')
-
-        bewit = genBewit()
-
-        if not bewit:
-            raise exceptions.TaskclusterFailure('Did not receive a bewit')
-
-        u = urllib.parse.urlparse(requestUrl)
-        return urllib.parse.urlunparse((
-            u.scheme,
-            u.netloc,
-            u.path,
-            u.params,
-            u.query + 'bewit=%s' % bewit,
-            u.fragment,
-        ))
+class RuntimeClient(BaseClient):
```

Comment:
I think it's sort of weird for this class to exist in the client.py file, when it's not the default used in client.py to export.  Can you maybe move this and `createApiClient` into a new file, maybe taskcluster/runtimeclient.py then import createApiClient and add it to this module's globals and __all__?